### PR TITLE
update keeper to accept gas price=0; default pending to latest

### LINKED ISSUE #242: web3.eth.getblock error "failed to get block bloom mapping"
__System info:__ [Include Ethermint commit, operating system name, and other relevant details]
Commit: 35e7a98ab2947f7a95c1bacb9262d252347c918b
MacOS

__Steps to reproduce:__

1. deploy ethermint chain 
2. run `npm run testDeploy` from github.com/chainsafe/ethermint-scripts

__Expected behavior:__ 
```
curl -X POST \
   --data '{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x5b", true],"id":1}' \
   -H "Content-Type: application/json" \
   http://localhost:8545

{"jsonrpc":"2.0","id":1,"result":{"difficulty":null,"extraData":null,"gasLimit":"0xffffffff","gasUsed":"0x47b760","hash":"0x496021a74bad57f9ff8dba2481817c08d6d253846e40127023f2bd94d51b5a39","logsBloom":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000200000000000020002000000000000000000000000000000000000000000000000000000000000000200000000000000000000000001000008000000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000000000000000400000080000000000000000000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000100000000800000000000000000000000000000000","miner":"0x0000000000000000000000000000000000000000","nonce":null,"number":"0x5b","parentHash":"0x9530919961404a4b8ea5be71eacf7726322ead4ac0ebac810c37c654c052e374","sha3Uncles":null,"size":"0x369","stateRoot":"0xa8486d186e1dc1439070664b40805fd8e271c24700492b0785eba2859f5b1c30","timestamp":"0x5e87b49d","totalDifficulty":null,"transactions":[{"blockHash":"0x496021a74bad57f9ff8dba2481817c08d6d253846e40127023f2bd94d51b5a39","blockNumber":"0x5b","from":"0xb990eefd1c6c2168609c561b526697ac34339cbd","gas":"0x72bf0","gasPrice":"0xa","hash":"0xb937a45f150b7e54f8bb3a42d9e853724f5016e4b8c81c9c0512f10d3260b78a","input":"0x221564090000000000000000000000000000000000000000000000000000000000000000","nonce":"0xc","to":"0xcdb226dc2f37a8c3938ae7adc371450981932ca8","transactionIndex":"0x0","value":"0x0","v":"0x34","r":"0x413f7ab557d98eff4b65fc82759539464432e8f0648922c73a33a27f69a2145c","s":"0x778168465541f1344b2f7d10b240bfe90527e0eb345916191e539975409af07c"}],"transactionsRoot":"0x95f343aac0dc7b107294a9dcf326014fa100ee8b27c8ebb77c6c4af6fc16c546","uncles":null}}
```

__Actual behavior:__ [What actually happened]

```
 block number: 91
(node:74627) UnhandledPromiseRejectionWarning: Error: Returned error: {"codespace":"sdk","code":1,"message":"failed to get block bloom mapping: block with bloombits [] not found"}
    at Object.ErrorResponse (/Users/letterj/workshop/chainsafe/ethermint-scripts/web3/node_modules/web3-core-helpers/src/errors.js:29:16)
    at /Users/letterj/workshop/chainsafe/ethermint-scripts/web3/node_modules/web3-core-requestmanager/src/index.js:140:36
    at XMLHttpRequest.request.onreadystatechange (/Users/letterj/workshop/chainsafe/ethermint-scripts/web3/node_modules/web3-providers-http/src/index.js:96:13)
    at XMLHttpRequestEventTarget.dispatchEvent (/Users/letterj/workshop/chainsafe/ethermint-scripts/web3/node_modules/xhr2-cookies/dist/xml-http-request-event-target.js:34:22)
    at XMLHttpRequest._setReadyState (/Users/letterj/workshop/chainsafe/ethermint-scripts/web3/node_modules/xhr2-cookies/dist/xml-http-request.js:208:14)
    at XMLHttpRequest._onHttpResponseEnd (/Users/letterj/workshop/chainsafe/ethermint-scripts/web3/node_modules/xhr2-cookies/dist/xml-http-request.js:318:14)
    at IncomingMessage.<anonymous> (/Users/letterj/workshop/chainsafe/ethermint-scripts/web3/node_modules/xhr2-cookies/dist/xml-http-request.js:289:61)
    at IncomingMessage.emit (events.js:323:22)
    at endReadableNT (_stream_readable.js:1204:12)
    at processTicksAndRejections (internal/process/task_queues.js:84:21)
(node:74627) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:74627) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

__Additional info:__ [Include gist of relevant config, logs, etc.]
All unit tests pass

### LINKED ISSUE #247: block logsBloom doesn't match block transactions
__System info:__ [Include Ethermint commit, operating system name, and other relevant details]

__Steps to reproduce:__

1. send a transaction that creates some logs
2. query for the block with this transaction
3. the block that includes this transaction may not have the correct logsBloom, but the block before/after it might contain the logs bloom for that block, but no transactions

overall it looks like the bloom is getting put in the wrong block.

__Expected behavior:__ [What you expected to happen]

bloom matches the block transactions

__Actual behavior:__ [What actually happened]

__Additional info:__ [Include gist of relevant config, logs, etc.]

related to #242

### LINKED ISSUE #298: cannot publish aragon app: pending queries not implemented
__System info:__ [Include Ethermint commit, operating system name, and other relevant details]

__Steps to reproduce:__

1. inside `aragon-react-boilerplate`:
```
$ npx buidler publish major --network default
main     | Applying version bump major, next version: 1.0.0
main     | Deploying new implementation contract
All contracts have already been compiled, skipping compilation.
An unexpected error occurred:

Error: Returned error: invalid argument 1: pending queries not implemented    at PromiEvent (/home/elizabeth/aragon-react-boilerplate/node_modules/@nomiclabs/truffle-contract/lib/promievent.js:13:30)
    at /home/elizabeth/aragon-react-boilerplate/node_modules/@nomiclabs/truffle-contract/lib/execute.js:223:26
    at Function.new (/home/elizabeth/aragon-react-boilerplate/node_modules/@nomiclabs/truffle-contract/lib/contract/constructorMethods.js:57:53)
    at _deployMainContract (/home/elizabeth/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/src/tasks/publish/index.ts:319:46)
```

__Expected behavior:__ [What you expected to happen]

__Actual behavior:__ [What actually happened]

__Additional info:__ [Include gist of relevant config, logs, etc.]

### PR DESCRIPTION
**Pull request by noot**
closes #298 and closes #247 and closes #242 
related to #297 

## Description

- default "pending" block queries to latest
- accept gas price = 0 
- set block bloom and hash mappings for current height, previously, the `getBlockByNumber(num, "latest")` call would attempt to get the bloom for the next block that hadn't been executed yet

```
 ETHERMINT_NODE_HOST=http://localhost:8545 ETHERMINT_INTEGRATION_TEST_MODE=stable go test ./tests/... -test.v -test.run ^TestBlockBloom
```

instructions for testing by deploying aragon boilerplate app:

first start ethermint node + rpc client

in another terminal, inside ChainSafe/aragonOS:
```
node scripts/deploy-notruffle.js 
```

in another terminal:
```
git clone https://github.com/aragon/aragon-react-boilerplate
cd aragon-react-boilerplate
```

inside arapp.json, change the default network to "localhost" and the "registry" to the ENS address deployed in the previous step
```
    "default": {
      "registry": "0xdfE3849648022Ed5D97Fd475c2964F1eE7a57776",
      "appName": "placeholder-app-name.aragonpm.eth",
      "network": "localhost"
    },
```

inside buidler.config.json, change localhost to:
```
    localhost: {
      from: '0x79A5F4B6c475a46Ffe2d919Da51f09235A61e398',
      accounts: ['0xFD4464657DCBA2DC582CD0A896934C52C55AC7A8A4558A81AFFF1F135507CFCB'],
      url: 'http://localhost:8545',
      chainId: 8,
      gas: 6700000,
    },
```

the private key in the `accounts` field is from `emintcli keys unsafe-export-eth-key mykey`, the `from` address is the `Owner` from deploy script step

in another terminal, run `ipfs daemon` (assuming you have ipfs installed)

finally, inside `aragon-react-boilerplate`:
` npx buidler publish major --network default`

you should see:
```
main     | Applying version bump major, next version: 1.0.0
main     | Deploying new implementation contract
All contracts have already been compiled, skipping compilation.
main     | Implementation contract deployed
main     | New implementation contract address: 0x24829c55d69d612300A36dfb7f6827ba9079eB98
main     | Running app build script
main     | Generating Aragon app artifacts
main     | Uploading release assets to IPFS...
(node:5510) ExperimentalWarning: Readable[Symbol.asyncIterator] is an experimental feature. This feature could change at any time
main     | Release assets uploaded to IPFS: QmcYSke7NXmEANGokRtQ9PiMCjzX7q6okdaUZWz6EquD7E
main     | 
main     |   Deploy new repo to registry aragonpm.eth
main     | 
main     |   App name:          placeholder-app-name.aragonpm.eth
main     |   Initial version:   1.0.0
main     |   Manager address:   0x79A5F4B6c475a46Ffe2d919Da51f09235A61e398
main     |   Contract address:  0x24829c55d69d612300A36dfb7f6827ba9079eB98
main     |   ContentURI:        QmcYSke7NXmEANGokRtQ9PiMCjzX7q6okdaUZWz6EquD7E
main     | 
main     |   http://localhost:8080/ipfs/QmcYSke7NXmEANGokRtQ9PiMCjzX7q6okdaUZWz6EquD7E
main     | 
main     | 
main     |   Tx sent
main     | 
main     |   Tx hash:  0xccb10126075167b3daeb81a7cb314188671fe9115131264d2cd00188af057a43
main     |   
```

after this it hangs because in ethermint there is an error:
```
I[2020-05-14|20:14:28.933] Rejected bad transaction                     module=mempool tx=9B3C1B68D081621C37292DDC195AA7B4B89AC2B94C10495A8476CD9F63B02897 peerID= res="&{CheckTx:code:3 log:\"invalid nonce; got 16, expected 15: invalid sequence\" gas_used:3894 codespace:\"sdk\" }" err=null
```
______

For contributor use:

- [x] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [x] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Comment by fedekunze** [comment]
```
npx buidler publish major --network default                                                                                                 
main     | Applying version bump major, next version: 1.0.0
main     | Deploying new implementation contract
All contracts have already been compiled, skipping compilation.
main     | Implementation contract deployed
main     | New implementation contract address: 0x6e751A131d7514eDE0c7EE975408e68F154Fd9e3
main     | Running app build script
main     | Generating Aragon app artifacts
main     | Uploading release assets to IPFS...
(node:86539) ExperimentalWarning: Readable[Symbol.asyncIterator] is an experimental feature. This feature could change at any time
main     | Release assets uploaded to IPFS: QmcVh8bQo283E4fCxPt5LfjhM7QmgRG64PtbjJyEGdsjHZ
main     | 
main     |   Deploy new repo to registry aragonpm.eth
main     | 
main     |   App name:          placeholder-app-name.aragonpm.eth
main     |   Initial version:   1.0.0
main     |   Manager address:   0xd3db233cd5487e2f2781021343dfbf1fc76985e6
main     |   Contract address:  0x6e751A131d7514eDE0c7EE975408e68F154Fd9e3
main     |   ContentURI:        QmcVh8bQo283E4fCxPt5LfjhM7QmgRG64PtbjJyEGdsjHZ
main     | 
main     |   https://ipfs.eth.aragon.network/ipfs/QmcVh8bQo283E4fCxPt5LfjhM7QmgRG64PtbjJyEGdsjHZ
main     | 
main     | 
main     |   Tx sent
main     | 
main     |   Tx hash:  0xac42201683a84c00978ff602185a7bf4e73be819387bb0d1b5ab5b09603db918
main     |   
main     | 
main     |   Tx mined
main     |   
main     |   Status:        Success
main     |   Block number:  480
main     |   Gas used:      1063623
main     | 
```

**Review by fedekunze — APPROVED** [review]
tested acknowledge  :+1: Great work @noot!

File: tests/rpc_test.go [hunk 1]
```diff
@@ -590,3 +599,41 @@ func TestEth_PendingTransactionFilter(t *testing.T) {
...
+	require.NoError(t, err)
+
+	lb := hexToBigInt(t, block["logsBloom"].(string))
+	require.NotEqual(t, big.NewInt(0), lb)
+	require.Equal(t, hash.String(), block["transactions"].([]interface{})[0])
+}
+
+func TestBlockBloom_Hash(t *testing.T) {
+	t.Skip()
+	// TODO: get this to work
```
Comment (fedekunze) [inline tests/rpc_test.go:624]: we can fix this in another PR

**Comment by thomasmodeneis** [comment]
I managed to make it run after few tries:

```
➜  aragon-react-boilerplate git:(master) ✗ npx buidler publish major --network default
Error BDLR109: Cannot connect to the network default.
Please make sure your node is running, and check your internet connection and networks config

For more info go to https://buidler.dev/BDLR109 or run Buidler with --show-stack-traces
➜  aragon-react-boilerplate git:(master) ✗ npx buidler publish major --network default
Error BDLR109: Cannot connect to the network default.
Please make sure your node is running, and check your internet connection and networks config

For more info go to https://buidler.dev/BDLR109 or run Buidler with --show-stack-traces
➜  aragon-react-boilerplate git:(master) ✗ npx buidler publish major --network localhost
An unexpected error occurred:

Error: Cannot convert string to buffer. toBuffer only supports 0x-prefixed hex strings and this string was given: B123B03EAB35D1072B18DE51F98D49BB255AB43E409F9FE775FE1333913678F8
    at exports.toBuffer (/opt/chainsafe/aragon-react-boilerplate/node_modules/@nomiclabs/buidler/node_modules/ethereumjs-util/src/bytes.ts:78:15)
    at hexPrivateKeys.map (/opt/chainsafe/aragon-react-boilerplate/node_modules/@nomiclabs/buidler/src/internal/core/providers/accounts.ts:37:49)
    at Proxy.map (<anonymous>)
    at createLocalAccountsProvider (/opt/chainsafe/aragon-react-boilerplate/node_modules/@nomiclabs/buidler/src/internal/core/providers/accounts.ts:37:38)
    at wrapEthereumProvider (/opt/chainsafe/aragon-react-boilerplate/node_modules/@nomiclabs/buidler/src/internal/core/providers/construction.ts:97:18)
    at Object.createProvider (/opt/chainsafe/aragon-react-boilerplate/node_modules/@nomiclabs/buidler/src/internal/core/providers/construction.ts:65:10)
    at Environment.lazy_1.lazyObject (/opt/chainsafe/aragon-react-boilerplate/node_modules/@nomiclabs/buidler/src/internal/core/runtime-environment.ts:76:14)
    at getRealTarget (/opt/chainsafe/aragon-react-boilerplate/node_modules/@nomiclabs/buidler/src/internal/util/lazy.ts:80:22)
    at Object.get (/opt/chainsafe/aragon-react-boilerplate/node_modules/@nomiclabs/buidler/src/internal/util/lazy.ts:146:26)
    at Web3HTTPProviderAdapter._sendJsonRpcRequest (/opt/chainsafe/aragon-react-boilerplate/node_modules/@nomiclabs/buidler-web3/src/web3-provider-adapter.ts:80:43)
    at util_1.default.callbackify (/opt/chainsafe/aragon-react-boilerplate/node_modules/@nomiclabs/buidler-web3/src/web3-provider-adapter.ts:46:35)
    at util.js:356:13
    at Web3HTTPProviderAdapter.send (/opt/chainsafe/aragon-react-boilerplate/node_modules/@nomiclabs/buidler-web3/src/web3-provider-adapter.ts:46:64)
    at /opt/chainsafe/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/node_modules/ethers/providers/web3-provider.js:69:19
    at new Promise (<anonymous>)
    at Web3Provider.send (/opt/chainsafe/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/node_modules/ethers/providers/web3-provider.js:62:16)
    at Web3Provider.JsonRpcProvider.perform (/opt/chainsafe/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/node_modules/ethers/providers/json-rpc-provider.js:302:29)
    at /opt/chainsafe/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/node_modules/ethers/providers/base-provider.js:826:34
➜  aragon-react-boilerplate git:(master) ✗ npx buidler publish major --network localhost
An unexpected error occurred:

Error: failed to simulate tx: account cosmos10xjlfdkywkjxll3djxw628cfyddxrcucndrcfs does not exist: unknown address
    at /opt/chainsafe/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/node_modules/ethers/providers/web3-provider.js:76:35
    at cb (util.js:353:39)
    at process._tickCallback (internal/process/next_tick.js:63:19)
➜  aragon-react-boilerplate git:(master) ✗ npx buidler publish major --network localhost
main     | Applying version bump major, next version: 1.0.0
Error in plugin @aragon/buidler-aragon: Registry aragonpm.eth does not exist

For more info run Buidler with --show-stack-traces
➜  aragon-react-boilerplate git:(master) ✗ npx buidler publish major --network default  
Error BDLR109: Cannot connect to the network default.
Please make sure your node is running, and check your internet connection and networks config

For more info go to https://buidler.dev/BDLR109 or run Buidler with --show-stack-traces
➜  aragon-react-boilerplate git:(master) ✗ npx buidler publish major --network default
main     | Applying version bump major, next version: 1.0.0
An unexpected error occurred:

Error: invalid input argument (arg="", reason="invalid address", value=null, version=4.0.46)
    at Object.throwError (/opt/chainsafe/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/node_modules/ethers/errors.js:76:17)
    at _FunctionDescription.encode (/opt/chainsafe/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/node_modules/ethers/utils/interface.js:85:20)
    at /opt/chainsafe/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/node_modules/ethers/contract.js:142:30
    at process._tickCallback (internal/process/next_tick.js:68:7)
➜  aragon-react-boilerplate git:(master) ✗ npx buidler publish major --network default
main     | Applying version bump major, next version: 1.0.0
An unexpected error occurred:

Error: invalid input argument (arg="", reason="invalid address", value=null, version=4.0.46)
    at Object.throwError (/opt/chainsafe/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/node_modules/ethers/errors.js:76:17)
    at _FunctionDescription.encode (/opt/chainsafe/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/node_modules/ethers/utils/interface.js:85:20)
    at /opt/chainsafe/aragon-react-boilerplate/node_modules/@aragon/buidler-aragon/node_modules/ethers/contract.js:142:30
    at process._tickCallback (internal/process/next_tick.js:68:7)
➜  aragon-react-boilerplate git:(master) ✗ npx buidler publish major --network default
main     | Applying version bump major, next version: 1.0.0
main     | Deploying new implementation contract
All contracts have already been compiled, skipping compilation.
main     | Implementation contract deployed
main     | New implementation contract address: 0xa25FdbB115B5bE401A0232F24c6237E4CCFBb836
main     | Running app build script
main     | Generating Aragon app artifacts
main     | Uploading release assets to IPFS...
main     | Release assets uploaded to IPFS: QmYo2PwywDsT9mtTK9M3W99CFkrQiUoYqJZdK7pT2Zs4p5
Error BDLR110: Network connection timed-out.
Please check your internet connection and networks config
```

For more info go to https://buidler.dev/BDLR110 or run Buidler with --show-stack-traces
```
➜  aragon-react-boilerplate git:(master) ✗ npx buidler publish major --network default
main     | Applying version bump major, next version: 1.0.0
main     | Deploying new implementation contract
All contracts have already been compiled, skipping compilation.
Error BDLR110: Network connection timed-out.
Please check your internet connection and networks config

success:
main     | Applying version bump major, next version: 1.0.0
main     | Deploying new implementation contract
All contracts have already been compiled, skipping compilation.
main     | Implementation contract deployed
main     | New implementation contract address: 0x10f2F183B3Fe0EBA6aC4a0b3bB88C3A07910956E
main     | Running app build script
main     | Generating Aragon app artifacts
main     | Uploading release assets to IPFS...
main     | Release assets uploaded to IPFS: QmYo2PwywDsT9mtTK9M3W99CFkrQiUoYqJZdK7pT2Zs4p5
main     | 
main     |   Deploy new repo to registry aragonpm.eth
main     | 
main     |   App name:          placeholder-app-name.aragonpm.eth
main     |   Initial version:   1.0.0
main     |   Manager address:   0x4568EE0a69614af61382bB0036aEBF18B796E7A2
main     |   Contract address:  0x10f2F183B3Fe0EBA6aC4a0b3bB88C3A07910956E
main     |   ContentURI:        QmYo2PwywDsT9mtTK9M3W99CFkrQiUoYqJZdK7pT2Zs4p5
main     | 
main     |   https://ipfs.eth.aragon.network/ipfs/QmYo2PwywDsT9mtTK9M3W99CFkrQiUoYqJZdK7pT2Zs4p5
main     | 
main     | 
main     |   Tx sent
main     | 
main     |   Tx hash:  0x24180d7f044c25275b34f0ddecb0f9fc986dddede21f3bca71365dd9215468cd
main     |   
main     | 
main     |   Tx mined
main     |   
main     |   Status:        Success
main     |   Block number:  589
main     |   Gas used:      1063619
main     | 
```

**Pull request merged**